src/containers/Search/Search.tsx

```diff
@@ -0,0 +1,143 @@
+  };
+
+  const handleProductSearch = async (pastSearch?: string) => {
+    if (updateRequestStatusCall) {
+      setShowPastSearch(false);
+
+      if (pastSearch) {
+        setSearchInputValue(pastSearch);
+      }
+
+      const lowerCaseSearchValue = (pastSearch || searchInputValue)?.toLowerCase();
+      const { data } = await updateRequestStatusCall(
+        () =>
+          baseApi.getProducts({
+            "filter[name]": lowerCaseSearchValue,
+          }),
+        URLNames.products
+      );
+
+      setSearchedProducts(data.data);
+
+      if (!pastSearch) {
+        const rawPastSearches = await AsyncStorage.getItem(PAST_SEARCH_STORAGE_KEY);
+        let updatedPastSearches: string[] = [];
+        if (rawPastSearches) {
+          updatedPastSearches = JSON.parse(rawPastSearches);
+
+          if (updatedPastSearches.length >= MAX_PAST_SEARCHES_STORED) {
+            updatedPastSearches.pop();
+          }
+        }
+
+        if (updatedPastSearches.indexOf(lowerCaseSearchValue) === -1 && lowerCaseSearchValue.length) {
+          updatedPastSearches.unshift(lowerCaseSearchValue);
+        }
+
+        await AsyncStorage.setItem(PAST_SEARCH_STORAGE_KEY, JSON.stringify(updatedPastSearches));
+
+        setPastSearches(updatedPastSearches);
+      }
+    }
+  };
+
+  useEffect(() => {
+    AsyncStorage.getItem(PAST_SEARCH_STORAGE_KEY).then((stringifiedPastSearches) => {
+      if (stringifiedPastSearches) {
+        try {
+          setPastSearches(JSON.parse(stringifiedPastSearches));
+        } catch (error) {
+          console.log(error);
+        }
+      }
+    });
+  }, []);
+
+  useEffect(() => {
+    if (inputRef) {
+      inputRef.current?.focus();
+    }
+
+    return () => {
+      dispatch?.({
+        type: URLNames.products,
+        payload: {
+          status: RequestStatus.RESET,
+        },
+      });
+    };
+  }, [dispatch]);
+  return (
+    <ScrollView style={[styles.searchContainer]}>
+      <Input
+        ref={inputRef}
+        value={searchInputValue}
+        onChange={handleInputChange}
+        onStartAdormentPress={() => handleProductSearch()}
+        startAdorment={<SearchIcon />}
+        isLoading={requestStatuses?.[URLNames.products].status === RequestStatus.REQUEST}
+      />
+      {shouldShowPastSearch && (
+        <View style={[styles.pastSearchesContainer]}>
+          {pastSearches.map((pastSearch) => {
+            const regexp = new RegExp(searchInputValue.toLocaleLowerCase(), "ig");
+            if (pastSearch.match(regexp)) {
+              return (
+                <TouchableOpacity
+                  style={[styles.pastSearchItem]}
+                  key={pastSearch}
+                  onPress={() => handleProductSearch(pastSearch)}
+                >
+                  <Typography>{pastSearch}</Typography>
+                </TouchableOpacity>
+              );
+            }
+          })}
+        </View>
+      )}
+      <View style={[styles.productsContainer]}>
```

Comment:
for handling a list of data use appropriate components for example FlatList

Reply:
fixed